### DISCUSSION TITLE
Paddle OCR fine-tuning to customized dataset

### DISCUSSION THREAD

**Discussion by Adhitya1996** [PaddlePaddle/PaddleOCR#13819] [Q&A]
Dear Paddle Community,

I am currently performing fine-tuning on text detection for paddle OCR on customized dataset. I have shared my YML file for your perusal.

1. We need inference model for both text detection as well as recognition. Currently I am performing fine-tuning with en_PP-OCRv3_det from the link: https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/doc/doc_en/models_list_en.md to perform fine-tuning on text detection. Now which one of the files is the final fine-tuned model for text detection?
![image](https://github.com/user-attachments/assets/e48bbf7a-595d-4fb8-a678-cba0202ce99b)
![image](https://github.com/user-attachments/assets/bb9303e2-8e97-4ee6-b185-19238202aac5)

Meaning which of the files do I have to use to convert the fine-tuned text detection model into a inference model? Which python script do I have to use for it?

2. What is the difference between best_accuracy.pdparams and student.pdparams that we get from trained model of en_pp-ocrv3_det?
3. Why for text detection we are using student.pdparams and best_accuracy from trained model of en_pp-ocrv3_rec for text recognition?

Apologies for questions but I request guidance at your earliest convenience!

Best Regards
Adhitya

**Comment by Adhitya1996** [comment]
Global:
  debug: false
  use_gpu: true
  epoch_num: 500
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: C:/Users/Adhi/paddle_fine_tuning/OCRv3_fine-tuned
  save_epoch_step: 50
  eval_batch_step:
  - 0
  - 400
  cal_metric_during_train: false
  pretrained_model: C:/Users/Adhi/paddle_fine_tuning/en_PP-OCRv3_det_distill_train/student.pdparams
  checkpoints: null
  save_inference_dir: null
  use_visualdl: false
  #infer_img: doc/imgs_en/img_10.jpg
  save_res_path: C:/Users/Adhi/paddle_fine_tuning/OCRv3_fine-tuned/OCRv3.txt
  distributed: true
  d2s_train_image_shape: [3, -1, -1]
  amp_dtype: bfloat16

Architecture:
  name: DistillationModel
  algorithm: Distillation
  model_type: det
  Models:
    Student:
      pretrained:
      model_type: det
      algorithm: DB
      Transform: null
      Backbone:
        name: MobileNetV3
        scale: 0.5
        model_name: large
        disable_se: true
      Neck:
        name: RSEFPN
        out_channels: 96
        shortcut: True
      Head:
        name: DBHead
        k: 50
    Student2:
      pretrained:
      model_type: det
      algorithm: DB
      Transform: null
      Backbone:
        name: MobileNetV3
        scale: 0.5
        model_name: large
        disable_se: true
      Neck:
        name: RSEFPN
        out_channels: 96
        shortcut: True
      Head:
        name: DBHead
        k: 50
    Teacher:
      freeze_params: true
      return_all_feats: false
      model_type: det
      algorithm: DB
      Backbone:
        name: ResNet_vd
        in_channels: 3
        layers: 50
      Neck:
        name: LKPAN
        out_channels: 256
      Head:
        name: DBHead
        kernel_list: [7,2,2]
        k: 50

Loss:
  name: CombinedLoss
  loss_config_list:
  - DistillationDilaDBLoss:
      weight: 1.0
      model_name_pairs:
      - ["Student", "Teacher"]
      - ["Student2", "Teacher"]
      key: maps
      balance_loss: true
      main_loss_type: DiceLoss
      alpha: 5
      beta: 10
      ohem_ratio: 3
  - DistillationDMLLoss:
      model_name_pairs:
      - ["Student", "Student2"]
      maps_name: "thrink_maps"
      weight: 1.0
      model_name_pairs: ["Student", "Student2"]
      key: maps
  - DistillationDBLoss:
      weight: 1.0
      model_name_list: ["Student", "Student2"]
      balance_loss: true
      main_loss_type: DiceLoss
      alpha: 5
      beta: 10
      ohem_ratio: 3

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.001
    warmup_epoch: 2
  regularizer:
    name: L2
    factor: 5.0e-05

PostProcess:
  name: DistillationDBPostProcess
  model_name: ["Student"]
  key: head_out
  thresh: 0.3
  box_thresh: 0.6
  max_candidates: 1000
  unclip_ratio: 1.5

Metric:
  name: DistillationMetric
  base_metric_name: DetMetric
  main_indicator: hmean
  key: "Student"

Train:
  dataset:
    name: SimpleDataSet
    data_dir: C:/Users/Adhi/paddle_fine_tuning/det_images
    label_file_list: [C:/Adhi/Adhi/paddle_fine_tuning/det_images/train/train_det.txt]
    ratio_list: [1.0]
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - DetLabelEncode: null
    - CopyPaste:
    - IaaAugment:
        augmenter_args:
        - type: Fliplr
          args:
            p: 0.5
        - type: Affine
          args:
            rotate:
            - -10
            - 10
        - type: Resize
          args:
            size:
            - 0.5
            - 3
    - EastRandomCropData:
        size:
        - 960
        - 960
        max_tries: 50
        keep_ratio: true
    - MakeBorderMap:
        shrink_ratio: 0.4
        thresh_min: 0.3
        thresh_max: 0.7
    - MakeShrinkMap:
        shrink_ratio: 0.4
        min_text_size: 8
    - NormalizeImage:
        scale: 1./255.
        mean:
        - 0.485
        - 0.456
        - 0.406
        std:
        - 0.229
        - 0.224
        - 0.225
        order: hwc
    - ToCHWImage: null
    - KeepKeys:
        keep_keys:
        - image
        - threshold_map
        - threshold_mask
        - shrink_map
        - shrink_mask
  loader:
    shuffle: true
    drop_last: false
    batch_size_per_card: 1
    num_workers: 4

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: C:/Users/Adhi/paddle_fine_tuning/det_images
    label_file_list: [C:/Users/Adhi/paddle_fine_tuning/det_images/test/test_det.txt]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - DetLabelEncode: # Class handling label
      - DetResizeForTest:
      - NormalizeImage:
          scale: 1./255.
          mean: [0.485, 0.456, 0.406]
          std: [0.229, 0.224, 0.225]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: ['image', 'shape', 'polys', 'ignore_tags']
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 1 # must be 1
    num_workers: 2

**Comment by UserWangZz** [comment]
I think you should use the knowledge distillation training process to fine-tune the model, but if you do not need to perform knowledge distillation on the model, you should reselect the yaml configuration file.

**Reply by Adhitya1996** [reply]
I reselected and change the yaml file and I replied regarding it in slightly detailed manner in another chat of ours. Thanks!
